### DISCUSSION TITLE
Option to auto hide unpopulate foreign keys !

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#5135] [General]
Consider a Book entity:

```Typescript
@Entity()
class Book {

  @PrimaryKey()
  id!: number;

  @ManyToOne(() => User, { ref: true })
  author!: Ref<User>;

}

const book = await em.findOneOrFail(Book, 1);
//  Real result: book = { id: 1, author: 12 }
//  Expected result: book = { id : 1 }
```

**Real result**
book = { id: 1, author: 12 }

**Expected result**
book = { id: 1 }

Can you add an option for that. Actually I try below serializer on @ManyToOne decorator on MikroOrm v5:
```Typescript
 serializer: (value) => {
        if (value?.isInitialized() && wrap(value, true).__populated) return wrap(value).toObject();
        else return undefined;
    },
```
But it's quite hard when i upgrade to v6. Yeah, I looked into the source code of MikroOrm but it's hard for me to understand what exactly algorithm of serialize function. Hope that you can an option for that, thank you so much !

**Comment by B4nan** [comment]
Sounds like you should be using partial loading if this is what you want. Or why would you want to select the value in the first place if you don't want it in the response?

> But it's quite hard when i upgrade to v6

What is quite hard? I am not aware of any changes in that code snippet when it comes to v6.

**Reply by dungkaka** [reply]
>Or why would you want to select the value in the first place if you don't want it in the response?

I just want to hide it if it's unpopulated. If it was populated, it is still in response. I don't use partial loading because I need to handle in each function, many function use a entity, so it's good if can handle at entity level. Moreover, I is common behavior for my project (for all foreign key case)

>What is quite hard? I am not aware of any changes in that code snippet when it comes to v6.

Hmm, wrap(value, true).__populated always return undefined in v6, I don't know why? But checking base on "__populated" is just my temporary solution, I'm not sure it right?  Can you give me some hint or keyword for handle the serialize in this case, I will look into source code for more understand ?

**Reply by B4nan** [reply]
Population works purely based on populate hints now, you won't get a bool for an entity instance, as the value is dependent on the context = the same entity instance can be populated somewhere (e.g. on top level) while not populated elsewhere (e.g. nested). The `__populated` flag is a way to enforce that something should be serialized as populated (regardless of the context), it's not something you should read and do things based on it.

Maybe you could also use [`lazy: true`](https://mikro-orm.io/docs/defining-entities#lazy-scalar-properties), which makes the property not selected (so just like partial loading, but declarative). Although this is not very tested for relations, it was meant to be used on scalar properties originally.
